include/seqan/blast/blast_record.h

```diff
@@ -0,0 +1,384 @@
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Hannes Hauswedell <hannes.hauswedell@fu-berlin.de>
+// ==========================================================================
+// Module for handling NCBI Blast I/O and E-Value computation
+// ==========================================================================
+
+#ifndef SEQAN_EXTRAS_BLAST_BLAST_RECORD_H_
+#define SEQAN_EXTRAS_BLAST_BLAST_RECORD_H_
+
+namespace seqan
+{
+
+/*!
+ * @class BlastMatch
+ * @implements AssignableConcept
+ * @implements CopyConstructibleConcept
+ * @implements DefaultConstructibleConcept
+ * @implements EqualityComparableConcept
+ * @implements LessThanComparableConcept
+ * @headerfile <seqan/blast.h>
+ * @signature struct BlastMatch<TQId, TSId, TPos, TAlign> { ... };
+ * @brief An data structure to hold a blast match, also known as high-scoring segment pair (HSP)
+ *
+ * You should set the following members manually: @link BlastMatch::qId @endlink, @link BlastMatch::sId @endlink,
+ * @link BlastMatch::qLength @endlink, @link BlastMatch::sLength @endlink,
+ * @link BlastMatch::qFrameShift @endlink and @link BlastMatch::sFrameShift @endlink.
+ *
+ * If you then also set a valid @link BlastMatch::align @endlink you can
+ * let the other members be computed by
+ * @link BlastMatch#computeAlignmentStats @endlink and
+ * @link BlastMatch#computeBitScore @endlink, @link BlastMatch#computeEValue @endlink.
+ *
+ * @tparam TQId  Type of qId, defaults to std::string
+ * @tparam TSId  Type of sId, defaults to std::string
+ * @tparam TPos  Position type of the sequences, defaults to <tt>uint32_t</tt>
+ * @tparam TAlign Type of the @link Align @endlink member, defaults to
+ * <tt>Align<CharString, ArrayGaps></tt>
+ *
+ */
+
+template <typename TQId = std::string,
```

Comment:
In some situations you will want to change the types, e.g. when you have all IDs in a ConcatDirect StringSet, it makes much more sense to also use Infixes for the IDs, so you don't need to copy.